backend/coreapp/middleware.py

```diff
@@ -0,0 +1,68 @@
+from django.http.request import HttpRequest
+from django.utils.timezone import now
+from django.contrib import auth
+
+from .models import User, Profile
+import logging
+from rest_framework.request import Request as DRFRequest
+from typing import Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .github import GitHubUser
+
+class AnonymousUser(auth.models.AnonymousUser):
+    profile: Profile
+
+class Request(DRFRequest):
+    user: Union[User, AnonymousUser]
+    profile: Profile
+
+def disable_csrf(get_response):
+    def middleware(request: HttpRequest):
+        setattr(request, "_dont_enforce_csrf_checks", True)
+        return get_response(request)
+
+    return middleware
+
+def set_user_profile(get_response):
+    """
+    Makes sure that `request.profile` is always available, even for anonymous users.
+    """
+
+    def middleware(request: HttpRequest):
+        profile = None
+
+        if not request.user.is_anonymous:
+            try:
+                profile = request.user.profile
+            except User.profile.RelatedObjectDoesNotExist: # type: ignore
+                pass
+
+        if not profile:
+            id = request.session.get("anonymous_profile_id")
+            if not isinstance(id, int):
+                id = -1
```

Comment:
`.get(..., -1)`? (+ an `assert isinstance(id, int)` if mypy is complaining?)

Though, I would recommend not using `-1` as a sentinel value -- because this logic would fall apart if there were a Profile in the DB with id `-1`. Can this be refactored to avoid trying `Profile.objects.get(...)` if the anon id isn't in `request.session`?